### DISCUSSION TITLE
Replacement of variables in html used as template

### DISCUSSION THREAD

**Discussion by jquiros2** [gotenberg/gotenberg#1089] [Q&A] [answered]
Hi! 
I'm calling gotenberg from node-red and have been trying to do something for some time and I just now realize it might not be a feature, and just looking for a clear answer:

I'm using the forms/chromium/convert/html route and getting the html converted to pdf.
Now I'm trying to add variables in the html template with {{ }} and trying different ways to send the data.

I just backstepped a bit, and realized I might've been assuming it was possible.

I found one [reference](https://github.com/gotenberg/gotenberg/issues/404) to something similar but I don't understand if the answer to switch to html was because the html converter can replace, or because it's easier to replace in html before sending to gotenberg.

Is it possible to have variables in the html substituted on gotenberg's end?  Or should I just replace in the html before sending to Gotenberg?

Thanks in advance for clarification, and for this great tool!

p.s. editted once as my wording was not clear.

**Comment by gulien** [comment, marked as answer]
Hey @jquiros2 👋 

You should replace the HTML before sending it to Gotenberg 😉
